robot-core/src/test/java/org/obolibrary/robot/DiffOperationTest.java

```diff
@@ -53,18 +54,19 @@ public void testCompareModified() throws IOException, OWLOntologyCreationExcepti
     boolean actual = DiffOperation.compare(simple, simple1, writer);
     System.out.println(writer.toString());
     assertFalse(actual);
-    String expected = IOUtils.toString(this.getClass().getResourceAsStream("/simple1.diff"));
+    String expected =
+        IOUtils.toString(
+            this.getClass().getResourceAsStream("/simple1.diff"), Charset.defaultCharset());
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.

Reply:
@lillieMaiBauer Are you a person or a bot? This looks like an automated message, and your GitHub profile is pretty bare, so I can't tell.

This code is only used in our test suite to work around annoyances loading test files. There may be a better way to do this. But it should never be extended by any other class.

Reply:
Thank you for your quick reply. Since this is a test class, this problem is inapplicable.

Reply:
Ok, thanks.